### ISSUE TITLE
Add product type selection flow from wp-admin

### ISSUE THREAD

**Issue by jarekmorawski** [woothemes/woocommerce#41742]
Related to https://github.com/woocommerce/woocommerce/issues/41586

We will soon allow merchants to change the product type from within the new product editing experience. It makes sense to go one step further and let them choose it from within the Products list in wp-admin. We can use that opportunity to improve the UX by adding descriptions and icons and allowing merchants to duplicate a product while creating a new one. According to our research, it's how most of them create new products.

**Design**

New UX enabled

<img width="392" alt="image" src="https://github.com/woocommerce/woocommerce/assets/79307566/6b3cd7e7-39c8-44f4-935e-e885b3dd6051">

New UX disabled

<img width="363" alt="image" src="https://github.com/woocommerce/woocommerce/assets/79307566/94b6e076-9e43-470d-a408-a30147943f9a">

Duplicate (nice to have)

<img width="437" alt="image" src="https://github.com/woocommerce/woocommerce/assets/79307566/f81da589-4d44-4a5e-ad97-e58d8ee38bd8">

**Acceptance criteria**

- [ ] When the user clicks `Add` in the Products new or `Add new` in the wp-admin nav, we display the new modal.
- [ ] The modal displays a list of product types with icons, names, and descriptions.
- [ ] The list is extensible and displays product types added by third-party plugins.
- [ ] If the new UX is enabled, we split the product type list into two sections: those that are compatible with it, and those that can only be created with the classic experience.
- [ ] Plugins incompatible with the new UX have a subheading that reads Classic experience and has a help text. Clicking the info icon reveals a tooltip that reads: `These products can be created with the classic editing screen that isn’t optimized for commerce but offers advanced functionality and supports all extensions. Learn more`. `Learn more` links to `https://woo.com/document/managing-products/`.
- [ ] If the new UX is disabled, we all product types are displayed in a single list. 
- [ ] When the user hovers over an item, the name changes color to primary. We also display an `Add` button. The user can click the whole row or the button to continue.
- [ ] When the user checks the “Make it default” box, next time they click “Add” in the products screen, they’ll go straight to the form with the selected product type.

Duplicate (nice to have)

- [ ] We display two tabs below the modal's title. `Create new` displays a list of available product types. `Duplicate` displays a list of previously created products the user can duplicate in one click.
- [ ] In the ‘Duplicate’ tab, users can search for any products in their store.
- [ ] The list includes 10 items. The user can click `Load more` to load 20 more.
- [ ] In the list, we display a product image (or a placeholder if there isn’t any), name, price, and number of variations (if applicable).
- [ ] On hover, we display a `Duplicate` button. The user can click it or the whole list item to continue.
- [ ] Clicking the item or button takes the user to the product editing experience and creates a copy of the selected product.
- [ ] The copy has the same information, including the product type.
- [ ] If the selected product is of a type that the new UX doesn’t support yet, we take the user to the classic editor.

**Comment by jarekmorawski** [comment]
@j111q, do you have any feedback about the design or acceptance criteria? This issue is just a starting point and we can continue iterating on the flow to make sure it's scalable and makes sense in terms of ease of use, clarity, and extensibility. cc @pmcpinto

**Comment by pmcpinto** [comment]
Jarek, I think the goal was to avoid mentioning classic or beta in the selection step, regardless of the situation. Also, I don't 100% it's worth showing this flow when the new experience is deactivated.

**Comment by jarekmorawski** [comment]
> Jarek, I think the goal was to avoid mentioning classic or beta in the selection step, regardless of the situation.

Right. We can tweak the copy and change the heading to `Other` or something similar. Maybe even collapse the section to make the basic/supported product types more prominent?

> Also, I don't 100% it's worth showing this flow when the new experience is deactivated.

It could if we wanted to bring more visibility to the less-used product types. We could, however, use the same dropdown menu as in the new UX. This would save users a click because they'd jump straight to a specific product type instead of changing it manually from within the editor – similar to the task list flow in the Home screen.

**Comment by pmcpinto** [comment]
I'm going to close this one. It refers to the new editor that will be removed.
